File: lessons/193-how-to-debug-kubernetes-deployments-services-ingress/myapp/src/main/java/com/antonputra/Device.java
package com.antonputra;

public class Device {

    public String uuid;
    public String mac;
    public String firmware;

    public Device() {
    }

    public Device(String uuid, String mac, String firmware) {
        this.uuid = uuid;
        this.mac = mac;
        this.firmware = firmware;
    }
}

File: lessons/244-rust-vs-go-golang-performance-fastest-frameworks-postgresql/go-app/device/device.go
package device

import (
	"context"

	mon "github.com/antonputra/go-utils/monitoring"
	"github.com/antonputra/go-utils/util"
	"github.com/jackc/pgx/v5/pgxpool"
)

type Device struct {
	Id       int    `json:"id"`
	Mac      string `json:"mac"`
	Firmware string `json:"firmware"`
}

func (d *Device) Save(ctx context.Context, db *pgxpool.Pool, m *mon.Metrics, sql string) (err error) {
	err = db.QueryRow(ctx, sql, d.Mac, d.Firmware).Scan(&d.Id)
	return util.Annotate(err, "db.Exec failed")
}
